doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,246 @@
+ * scorecard will require a valid bundle image or on-disk bundle to determine what information will be passed to scorecard tests
+ * scorecard will determine test configurations by a YAML/DSL file that is either passed on the command line or included in the bundle contents
+ * scorecard will only run OLM tests if OLM manifests are found in the bundle
+ * scorecard command syntax will look as follows:
+   * operator-sdk scorecard --dsl <point-to-yaml> --bundle <bundle-path/image> -l <selector>
+
+### User Stories
+
+#### Story 1
+
+I am as an Operator developer, I'd like to be able to construct custom tests and execute them via the scorecard.
+
+#### Story 2
+
+Scorecard will include an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will be converted to the custom test format and therefore be externalized from the scorecard binary itself.
+
+#### Story 4
+
+Scorecard would process tests based on an on-disk directory or bundle image that contains the scorecard configuration.
+
+#### Story 5 
+
+- I am as an Operator developer, would like to be able to create tests using the same Assert syntax adopted by [Kuttl](https://github.com/kudobuilder/kuttl) and check its results aggregated within scorecard results. 
+
+**Implementation Notes**
+
+The Scorecard would support the creation of a Kuttl test image and the execute tests via Kuttl, however, it would be transparent for users since its results will be integrated into scorecard results.
+
+## Design Details
+
+### Glossary
+
+ * test image - this is a container image built by end-users to contain a custom test implementation
+ * test configuration (DSL) - this is a file that contains the test definitions and labels which determine what tests are run and how they are configured
+ * Custom Test Definition - A custom test is any sort of test logic required, written in any programming language that satifies the scorecard test requirements for output format and execution.
+
+#### Test Image
+
+A custom test is based on a container image, that image is created by
+the end-user using whatever tools they want to use.
+
+The test image must produce container log output that conforms to the scorecard Test Result output.  The output is an array of Test Results, so a test image could contain multiple tests if necessary.  Here is a sample of test output in the required format:
+```
+    [
+      {
+	      "name": "Spec fields with descriptors",
+	      "description": "All spec fields have matching descriptors in the CSV",
+	      "state": "fail",
+	      "suggestions": [
+		"Add a spec descriptor for size"
+	      ],
+	      "crname": "example-memcached"
+      }, 
+      {
+	      "name": "Some other test",
+	      "description": "Another test on the CSV",
+	      "state": "pass",
+	      "suggestions": [
+	      ],
+	      "crname": "example-memcached"
+      },
+    ]
```

Comment:
We discussed some changes to this output format, right? Remove the array, include just one test, and remove CR name?

Reply:
removed array in latest commit.